OutSystems/outsystems-datagrid-reactive
# Stories/rgridt 687

This PR is for...

[https://pp-datagridteam-dev.outsystemsenterprise.com/DataGridSampleReactive_RGRIDT687/SaveChanges](url)

### What was happening
* When the user was calling the MarkChangesAsSaved, we were cleaning the DirtyMarks and the ValidationMarks. 

### What was done
* I've changed the API and all the methods including the provider code, so we can choose if we should clean the validation marks or not, using a dynamic flag (boolean).

### Test Steps
1. Build a grid with a SetValidationStatus on the OnCellValueChange;
2. Change the validation status, _IsValid_ value to False;
3. On another button use a new action that uses **MarkChangesAsSaved** client action;
4. Config the value of _ClearInvalids_ to **_False_**
5. Call this action after changing the value on the previous column
6. The invalid status mark should continue on the cell;
7. Config the value of _ClearInvalids_ to **_True_**
8. Repeat the test
9. The invalid status mark shouldn't be displayed;

### Screenshots
(prefer animated gif)


### Checklist
